packages/mui-base/src/ButtonUnstyled/ButtonUnstyled.tsx

```diff
@@ -76,30 +82,26 @@ const ButtonUnstyled = React.forwardRef(function ButtonUnstyled<
     [setFocusVisible],
   );
 
-  const ownerState = {
+  const ownerState: ButtonUnstyledOwnerState = {
     ...props,
     active,
     focusVisible,
   };
 
-  const ButtonRoot: React.ElementType = component ?? components.Root ?? 'button';
-  const buttonRootProps = appendOwnerState(
+  const classes = useUtilityClasses(ownerState);
+
+  const buttonRootProps: WithOptionalOwnerState<ButtonUnstyledRootSlotProps> = appendOwnerState(
     ButtonRoot,
-    { ...other, ...componentsProps.root },
+    {
+      ...getRootProps(),
+      ...other,
+      ...componentsProps.root,
+      className: clsx(classes.root, className, componentsProps.root?.className),
```

Comment:
In the `MultiSelectUnstyled` the other is a bit different:
```jsx
  const buttonProps: WithOptionalOwnerState<MultiSelectUnstyledRootSlotProps<TValue>> =
    appendOwnerState(
      Button,
      {
        ...other,
        ...componentsProps.root,
        ...getButtonProps(),
        className: clsx(className, componentsProps.root?.className, classes.root),
      },
      ownerState,
    );
```

Should we normalize this?

Reply:
I think what we have in the `ButtonUnstyled` is correct, because we allow developers to override what's coming from the `getRootProps`

Reply:
Right! Thanks for noticing!